openstack/cdn/v1/services/requests.go

```diff
@@ -260,7 +261,7 @@ func Update(c *gophercloud.ServiceClient, idOrURL string, opts UpdateOpts) (r Up
 		b[i] = patch.ToCDNServiceUpdateMap()
 	}
 
-	resp, err := c.Request("PATCH", url, &gophercloud.RequestOpts{
+	resp, err := c.Request(context.Background(), "PATCH", url, &gophercloud.RequestOpts{
```

Comment:
To add cancellation capabilities to this `Update` function, create a `UpdateWithContext` clone and turn this into a convenience function that adds a `context.Background()`.

```Go
func Update(c *gophercloud.ServiceClient, idOrURL string, opts UpdateOpts) (r UpdateResult) {
	return UpdateWithContext(context.Background(), c, idOrURL, opts)
```